protocols/imap/src/main/java/org/apache/james/imap/processor/SearchProcessor.java

```diff
@@ -221,7 +203,27 @@ protected void doProcess(SearchRequest request, ImapSession session, String tag,
             session.setAttribute(SEARCH_MODSEQ, null);
         }
     }
-    
+
+    private Collection<Long> asResults(ImapSession session, boolean useUids, Collection<MessageUid> uids) {
+        if (useUids) {
+            return uids.stream()
+                .map(MessageUid::asLong)
+                .collect(Guavate.toImmutableList());
+        } else {
+            return uids.stream()
+                .map(uid -> session.getSelected().msn(uid))
+                .map(i -> (long) i)
```

Comment:
please avoid c-like cast syntax. you can write `Integer.valueOf(i).longValue()` to make it clear what you have as input, what as output (and also be resilient with msn() return type changes)